src/backend/opencl/convolve.cpp

```diff
@@ -85,5 +94,197 @@ INSTANTIATE(ushort ,   float)
 INSTANTIATE(short  ,   float)
 INSTANTIATE(uintl  ,   float)
 INSTANTIATE(intl   ,   float)
+#undef INSTANTIATE
+
+template<typename T, typename accT>
+Array<T> convolve2_unwrap(const Array<T>& signal,
+                          const Array<accT>& filter,
+                          const dim4 stride,
+                          const dim4 padding,
+                          const dim4 dilation)
+{
+    dim4 sDims = signal.dims();
+    dim4 fDims = filter.dims();
+
+    dim_t outputWidth  = 1 + (sDims[0] + 2 * padding[0] - (((fDims[0] - 1) * dilation[0]) + 1)) / stride[0];
+    dim_t outputHeight = 1 + (sDims[1] + 2 * padding[1] - (((fDims[1] - 1) * dilation[1]) + 1)) / stride[1];
+    Array<T> out = createValueArray<T>(dim4(outputWidth, outputHeight, fDims[3], sDims[3]), scalar<T>(0));
```

Comment:
We don't need a `createValueArray` here when we are actually setting `out` in line 138 from the output of another operation.